File: Data Structures & Algorithms/anagram-groups/submission-5.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
      unordered_map<string,vector<string>> res;

      for(auto i:strs){
        string key=i;
        sort(key.begin(),key.end());
        res[key].push_back(i);
      }
      vector<vector<string>> ans;
      for(auto &i:res){
        ans.push_back(i.second);
      }
      return ans;
    }
};

File: Data Structures & Algorithms/longest-common-prefix/submission-5.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        prefix=strs[0]
        for i in range(len(prefix)):
            char=prefix[i]
            for s in strs[1:]:
                if i>=len(s) or s[i]!=char:
                    return prefix[:i]
        return prefix